Core/src/main/java/com/example/core/extension/View.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.core.extension
+
+import android.animation.Animator
+import android.animation.ValueAnimator
+import android.graphics.drawable.Drawable
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.DecelerateInterpolator
+import android.widget.AdapterView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.appcompat.widget.AppCompatEditText
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.appcompat.widget.AppCompatSpinner
+import androidx.core.content.ContextCompat
+import com.squareup.picasso.Picasso
+import de.hdodenhof.circleimageview.CircleImageView
+
+private const val DURATION = 300
+
+fun View.visible() {
+    visibility = View.VISIBLE
+}
+
+fun View.invisible() {
+    visibility = View.INVISIBLE
+}
+
+fun View.gone() {
+    visibility = View.GONE
+}
+
+fun AppCompatImageView.load(resId: String) {
+    Picasso.get().load(resId).into(this)
+}
+
+fun CircleImageView.load(resId: String, centerCrop: Boolean = true, fit: Boolean = true) {
+    Picasso.get().load(resId).also { if (centerCrop) it.centerCrop() }.also { if (fit) it.fit() }
+        .into(this)
+}
+
+fun View.getImageDrawable(@DrawableRes id: Int): Drawable? = context.getImageDrawable(id)
+
+fun View.getResolvedColor(@ColorRes id: Int): Int = ContextCompat.getColor(context, id)
+
+fun AppCompatEditText.afterTextChanged(afterTextChanged: (String) -> Unit) {
+    this.addTextChangedListener(object : TextWatcher {
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            Log.i("ihek", "beforeTextChanged: ")
```

Comment:
Add proper tag name for log, "ihek" looking very bed.

Reply:
done